source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs

```diff
@@ -1,15 +1,24 @@
 ﻿using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System;
 
 namespace Pacioli.Lib.Models
 {
-    public class JournalEntry
+    public record JournalEntry
     {
         public JournalEntry()
         {
             Debits = new List<JournalEntryItem>();
             Credits = new List<JournalEntryItem>();
         }
 
+        public JournalEntry([NotNull] List<JournalEntryItem> debits, [NotNull] List<JournalEntryItem> credits)
+        {
+            if ((!debits?.Any() ?? false) || (!credits?.Any() ?? false))
```

Comment:
Maybe we should split this into an ArgumentNullException and ArgumentException thrower.

Reply:
Yeah that makes sense.